setup.py

```diff
@@ -63,118 +62,72 @@ def get_extras_require() -> Dict[str, List[str]]:
-            "plotly>=4.0.0",
+            "pillow",
+            "plotly>=4.0.0",  # optuna/visualization.
+            "scikit-learn",
             "scikit-learn>=0.24.2",
             "scikit-optimize",
-            "mlflow",
-        ],
-        "document": [
             "sphinx",
-            "sphinx_rtd_theme",
             "sphinx-copybutton",
             "sphinx-gallery",
             "sphinx-plotly-directive",
-            "pillow",
-            "matplotlib",
-            "scikit-learn",
-            "plotly>=4.0.0",  # optuna/visualization.
-            "pandas",
-            "lightgbm",
+            "sphinx_rtd_theme",
+            "thop",
             "torch==1.10.0",
-            "torchvision==0.11.1",
             "torchaudio==0.10.0",
-            "thop",
+            "torchvision==0.11.1",
         ],
-        "experimental": ["redis"],
-        "testing": [
-            "chainer>=5.0.0",
-            "cma",
-            "fakeredis",
-            "lightgbm",
-            "matplotlib>=3.0.0",
-            "mlflow",
-            "mpi4py",
-            "mxnet",
-            "pandas",
-            "plotly>=4.0.0",
-            "pytest",
-            "scikit-learn>=0.24.2",
-            "scikit-optimize",
-            "xgboost",
-            "tensorflow ; python_version>'3.6'",
-            "tensorflow-datasets",
-            "pytorch-ignite ; python_version<'3.10'",
-            "pytorch-lightning>=1.5.0,<1.6.0 ; python_version<'3.10'",
-            "skorch ; python_version<'3.10'",
-            "catalyst>=21.3 ; python_version<'3.10'",
-            "torch==1.10.0 ; sys_platform=='darwin' and python_version<'3.10'",
-            "torch==1.10.0+cpu ; sys_platform!='darwin' and python_version<'3.10'",
-            "torchvision==0.11.1 ; sys_platform=='darwin' and python_version<'3.10'",
-            "torchvision==0.11.1+cpu ; sys_platform!='darwin' and python_version<'3.10'",
-            "torchaudio==0.10.0; python_version<'3.10'",
-            # TODO(himkt): Remove `nltk` after solving
-            # https://github.com/allenai/allennlp/issues/5521
-            "nltk<3.6.6 ; python_version<'3.10'",
+        "integration": [
             # TODO(himkt): Remove upper bound constraint
             # after solving https://github.com/optuna/optuna/issues/3366
             "allennlp>=2.2.0,<2.9.1 ; python_version>'3.6' and python_version<'3.10'",
             "botorch>=0.4.0 ; python_version>'3.6' and python_version<'3.10'",
-            "fastai ; python_version>'3.6' and python_version<'3.10'",
-        ],
-        "tests": [
-            "fakeredis",
-            "pytest",
-        ],
-        "optional": [
-            "matplotlib>=3.0.0",  # optuna/visualization/matplotlib
-            "pandas",  # optuna/study.py
-            "plotly>=4.0.0",  # optuna/visualization.
-            "redis",  # optuna/storages/redis.py.
-            "scikit-learn>=0.24.2",
-            # optuna/visualization/param_importances.py.
-        ],
```

Comment:
`tests` and `optional` are always used simultaneously to make me think of simply unifying. @not522 

https://cs.github.com/optuna/optuna?q=tests%20path%3A%2F%5E.github%5C%2Fworkflows%5C%2F%2F

Reply:
I prefer to separate tests depending on whether they require additional libraries. E.g.,

```
pip install .[tests]
pytest tests -m "not optional"
pip install .[optional]
pytest tests -m "optional"
```

It will prevent accidentally importing optional libraries as default.

Reply:
[5ad5e76](https://github.com/optuna/optuna/pull/3517/commits/5ad5e7634813c0e1d37f0045c93263bc6c8b2581)
Restored `test` (plural -> singular for the consistency with other extras) and `optional`. 🙌